server/src/api/v1/users/auth-user.ts

```diff
@@ -1,28 +1,45 @@
 import {Request, Response} from 'express';
 import {wrapResponse} from '../../../functions/response-wrapper';
-import {UserData} from '../../../interfaces/users.interface';
-import {mapUser} from '../../../functions/map-users.func';
+import {UserLoginData} from '../../../interfaces/users.interface';
 import {User} from '../../../models/user.model';
 import * as bcrypt from 'bcryptjs';
 import {jwtSign} from '../../../functions/jwt-sign.func';
 
 export async function loginUser(req: Request, res: Response): Promise<Response> {
 
-    const incomingData: UserData = req.body;
-    const mappedIncomingData: UserData = await mapUser(incomingData);
+    const incomingData: UserLoginData = req.body;
 
-    let success = true;
+    if (incomingData.password === undefined || (incomingData.email === undefined && incomingData.username === undefined)){
```

Comment:
Kannst du nicht hier unsere requirdedFields funktion anwenden?

```suggestion
if(!requiredFields(data, ['email', 'password'] || !requiredFields(data, ['user', 'password']) {
```
(pseudo code, weiß den syntax nicht mehr)